### ISSUE TITLE
Binder omits a subset of module-level classes from google-genai's large types.py (symbols present in source + runtime)

### ISSUE THREAD

**Issue by gtstamant** [DetachHead/basedpyright#1841]
## Summary

basedpyright fails to bind a **subset** of the module-level classes in `google.genai.types` (from the `google-genai` package). The dropped names are reported as `"X" is unknown import symbol` / `"X" is not a known attribute of module "google.genai.types"`, even though they are ordinary module-level `class` definitions present in the source and importable at runtime. Other classes in the *same* file resolve fine. The class body of `GenerateContentConfig` is also truncated (later fields like `tools`/`tool_config` are treated as nonexistent, and two optional fields are read as required).

This looks like a binder/parse limit on a very large machine-generated file (`types.py` is ~830 KB / 22,465 lines), likely inherited from pyright.

## Minimal repro

```sh
pip install google-genai==2.9.0
```

`probe.py`:
```python
from google.genai import types

reveal_type(types.Content)        # resolves:   type[Content]
reveal_type(types.Tool)           # FAILS:      Unknown  +  "Tool" is not a known attribute of module
reveal_type(types.FunctionCall)   # FAILS:      Unknown
reveal_type(types.ToolConfig)     # FAILS:      Unknown
```

```sh
basedpyright probe.py
```

Also fails (truncated class body):
```python
from google.genai import types
# reportCallIssue: "No parameter named 'tools'" / "No parameter named 'tool_config'",
# plus spurious "Arguments missing for parameters 'response_mime_type', 'response_json_schema'"
types.GenerateContentConfig(tools=[], tool_config=None)
```

## Expected

`types.Tool`, `types.FunctionCall`, `types.ToolConfig`, `types.FunctionDeclaration`, `types.FunctionCallingConfig`, `types.FunctionCallingConfigMode` resolve to their classes, and `GenerateContentConfig` exposes all its fields (`tools`, `tool_config`, …) — as they do at runtime.

## Evidence that the source is correct (the checker, not the SDK, is wrong)

- **Source:** Python's own `ast.parse(types.py)` shows all of these are module-level `ClassDef`s, defined exactly once each (1,485 top-level statements total).
- **Runtime:** `from google.genai import types; types.Tool(...)` works; `GenerateContentConfig(tools=[{...}], tool_config={...})` constructs (pydantic coerces the dicts to `Tool`/`ToolConfig`).
- **Deterministic, no line-order pattern.** By definition line: `419` dropped · `1636` dropped · `1999` kept · `2299` kept · `4444–5817` dropped · `5898`, `8027` kept. Not a simple "first N survive" cutoff.
- It is a **binder-level** failure (the name is absent from the module symbol table — `unknown import symbol`), not a type-evaluation failure (which would keep the symbol and report its type as Unknown).

## Versions

- basedpyright **1.39.6** and **1.39.8** (both reproduce identically), based on pyright **1.1.410**
- google-genai **2.9.0** (latest)
- Python **3.14**, `typeCheckingMode = "strict"`

Happy to provide more diagnostics. If this is squarely an upstream pyright binder limit, glad to re-file at microsoft/pyright — flagging here first since I hit it via basedpyright.

**Comment by DetachHead** [comment]
i can't reproduce this. this issue description looks AI generated, can you confirm that this is a real issue you encountered and not something hallucinated by an AI? assuming it's a real issue, could you provide some more info please:

- what does your basedpyright config look like?
- are you running basedpyright from inside your venv?
- does it also happen in the language server? (for example in vscode)
- can you narrow down a repro that doesn't require installing a 3rd party package

**Comment by gtstamant** [comment]
hi,

- config: I'm using typeCheckingMode = "strict", pythonVersion = "3.14", reportUnnecessaryTypeIgnoreComment = "error", and the rest is the default strict set
- uv managed venv, basedpyright resolves google-genai. from it
- yes, have the same problem using CLI/vscode

The issue is with a gemini library file that is 22,465 lines long. I was able to reproduce a similar problem using this generator:

```python
lines = ["from dataclasses import dataclass", "", "@dataclass", "class Settings:"]
lines += [f"    field{i}: int | None = None" for i in range(32)]
lines += ["    tools: list[int] | None = None", "    tool_config: int | None = None", ""]
for k in range(2400):
    lines += ["@dataclass", f"class Filler{k}:"]
    lines += [f"    a{j}: int | None = None" for j in range(6)] + [""]
open("bigmodule.py", "w").write("\n".join(lines))
open("myprogram.py", "w").write(
    "from bigmodule import Settings\n"
    "Settings(tools=[1], tool_config=2)\n")
```

```
python3 make_repro.py
basedpyright myprogram.py
```

```
myprogram.py:2:10 - error: No parameter named "tools" (reportCallIssue)
myprogram.py:2:21 - error: No parameter named "tool_config" (reportCallIssue)
```

happy to help further if i can.

**Comment by DetachHead** [comment]
looks like this is the problem:

```
/projects/test/bigmodule.py: error: Code is too complex to analyze; reduce complexity by refactoring into subroutines or reducing conditional code paths (reportGeneralTypeIssues)
```
